Add custom templating to templating.md (#3592)
astronomer/docs: learn/templating.md

</TabItem>
</Tabs>

### Templating additional fields

> So far, we've detailed how to pass Jinja template syntax to Operators' templateable fields. But what if you need to template a field not listed in the Operator's `template_fields`?

Reviewer: ```suggestion
If you need to template a field that is not listed in the operator's `template_fields`, you can either set the `template_fields` attribute on a task or create a custom operator.
```

> There are two ways to work around this problem. Either set the template_fields attribute on a Task or create a custom Operator.

Reviewer: ```suggestion
```

> Here is how you would use each method to template the `cwd` field of the BashOperator.

Reviewer: ```suggestion
The following example shows how to use each method to template the `cwd` field of the BashOperator.
```

> 1. Set the template_fields attribute of the task itself:

Reviewer: Rather than having numbered steps here, since this is showing two different methods, rather than sequential steps, can we show this information in two tabs? You can see an example of how to implement in the previous section where we have tabs for TaskFlow API and Traditional syntax examples.

>    Once the Task is defined and assigned to a Python variable, you can reassign its template_fields. This allows jinja templating for any field not templated by default. 

Reviewer: ```suggestion
   Once the task is defined and assigned to a Python variable, you can reassign its template_fields. This allows Jinja templating for any field that is not templated by default. 
```

    from airflow.decorators import dag
    from airflow.operators.bash import BashOperator
    from airflow.utils.dates import days_ago

    @dag(schedule=None, start_date=days_ago(1))
    def templating_dag():
        bash_task = BashOperator(
            task_id="set_template_field_after_task_instantiation",
            bash_command="script.sh",
            cwd="/usr/local/airflow/{{ ds }}",
        )
        bash_task.template_fields = ("bash_command", "env", "cwd")

   templating_dag()
   ```

3. Create a custom Operator by subclassing

   ```python
    from airflow.decorators import dag
    from airflow.operators.bash import BashOperator
    from airflow.utils.dates import days_ago
    from typing import TYPE_CHECKING

    if TYPE_CHECKING:
        from collections.abc import Sequence

    class TemplatedBashOperator(BashOperator):
        template_fields: Sequence[str] = ("bash_command", "env", "cwd")

    @dag(schedule=None, start_date=days_ago(1))
    def templating_dag():
        bash_task = BashOperator(
            task_id="set_template_field_after_task_instantiation",
            bash_command="script.sh",
            cwd="/usr/local/airflow/{{ ds }}",
        )

> templating_dag()

Reviewer: This should be indented to match the rest of the code snippet. Because it isn't, it's treating this line as regular text, and starting a new code block for the following section. (you can check the deploy preview link on the PR to make sure it's rendering properly)

        {label: 'Create a custom Operator', value: 'operator'},
    ]}>
<TabItem value="task">

After defining a task and assigning it to a Python variable, you can modify its `template_fields` attribute. This allows you to enable Jinja templating for any field that is not templated by default.

Prefer this method when you need to template a field only once. 

```python
    from airflow.decorators import dag
    from airflow.operators.bash import BashOperator
    from airflow.utils.dates import days_ago

    @dag(schedule=None, start_date=days_ago(1))
    def templating_dag():
        bash_task = BashOperator(
            task_id="set_template_field",
            bash_command="script.sh",
            cwd="/usr/local/airflow/{{ ds }}",
        )
        bash_task.template_fields = ("bash_command", "env", "cwd")

    templating_dag()
```

</TabItem>
<TabItem value="operator">

Create a custom Operator by subclassing the desired Operator class and adding your desired field to `template_fields`.
In this example, TemplatedBashOperator is a new operator that inherits the behavior of BashOperator and allows Jinja templating of the `cwd` field.

> Prefer this method if you need to template a field repeatedly.

Reviewer: ```suggestion
This method is preferred if you need to template a field repeatedly.
```
